kernel/thread.c

```diff
@@ -123,6 +123,26 @@ int saved_always_on = k_enable_sys_clock_always_on();
 }
 #endif
 
+#ifdef CONFIG_THREAD_MONITOR
+void _impl_k_thread_name_set(const char *value)
+{
+	_current->name = value;
+}
+
+#ifdef CONFIG_USERSPACE
+Z_SYSCALL_HANDLER(k_thread_name_set, data)
+{
+	_impl_k_thread_name_set((void *)data);
```

Comment:
data is untrusted and could be something that is not a C string, or even a garbage pointer.
We also shouldn't be storing user-supplied data in the kernel object, it could be modified later unexpectedly.

You need to call z_user_string_alloc_copy() on the data parameter so that it is checked and copied into an allocation from the thread's resource pool (which must be inherited from its parent or set at some point with k_thread_resource_pool_assign() or k_thread_system_pool_assign())

Also, this API is not very flexible as it only works on the current thread, consider adding an explicit thread object argument.